addOns/commonlib/src/test/java/org/zaproxy/addon/commonlib/scanrules/ScanRuleMetadataUnitTest.java

```diff
@@ -44,44 +44,102 @@ void setUp() {
+solution: The solution
+references:
+  - https://example.com/reference-1
+  - https://example.com/reference-2
+category: INJECTION  # info_gather, browser, server, misc, injection
+risk: INFO  # info, low, medium, high
+confidence: LOW  # false_positive, low, medium, high, user_confirmed
+cweId: 123
+wascId: 456
+alertTags:
+  name1: value1
+  name2: value2
+otherInfo: Any other Info
+status: alpha
+codeLink: https://www.example.com/codelink
+helpLink: https://www.example.com/helplink
+alertRefOverrides:
+  12345-1:
+    name: Alert Ref Override 1
+  12345-2:
+    name: Alert Ref Override 2
+""";
         // When
         var metadata = ScanRuleMetadata.fromYaml(yaml);
         // Then
         assertThat(metadata.getId(), is(equalTo(12345)));
         assertThat(metadata.getName(), is(equalTo("Active Vulnerability Title")));
         assertThat(metadata.getDescription(), is(equalTo("Full description")));
         assertThat(metadata.getSolution(), is(equalTo("The solution")));
-        assertThat(metadata.getReferences(), contains("Reference 1", "Reference 2"));
+        assertThat(
+                metadata.getReferences(),
+                contains("https://example.com/reference-1", "https://example.com/reference-2"));
         assertThat(metadata.getCategory(), is(equalTo(Category.INJECTION)));
         assertThat(metadata.getRisk(), is(equalTo(Risk.INFO)));
         assertThat(metadata.getConfidence(), is(equalTo(Confidence.LOW)));
-        assertThat(metadata.getCweId(), is(equalTo(0)));
-        assertThat(metadata.getWascId(), is(equalTo(0)));
+        assertThat(metadata.getCweId(), is(equalTo(123)));
+        assertThat(metadata.getWascId(), is(equalTo(456)));
         assertThat(
                 metadata.getAlertTags(), is(equalTo(Map.of("name1", "value1", "name2", "value2"))));
         assertThat(metadata.getOtherInfo(), is(equalTo("Any other Info")));
         assertThat(metadata.getStatus(), is(equalTo(AddOn.Status.alpha)));
         assertThat(metadata.getCodeLink(), is(equalTo("https://www.example.com/codelink")));
         assertThat(metadata.getHelpLink(), is(equalTo("https://www.example.com/helplink")));
+        assertThat(metadata.getAlertRefOverrides().size(), is(equalTo(2)));
+        assertThat(
+                metadata.getAlertRefOverrides().get("12345-1").getName(),
+                is(equalTo("Alert Ref Override 1")));
+        assertThat(
+                metadata.getAlertRefOverrides().get("12345-2").getName(),
+                is(equalTo("Alert Ref Override 2")));
+    }
+
+    @Test
+    void shouldHandleAllAlertRefOverrideFields() {
+        // Given
+        String yaml =
+                """
+id: 12345
+name: Active Vulnerability Title
+alertRefOverrides:
+  12345-1:
+    name: Alert Ref 1
+    description: Full description
+    solution: The solution
+    references:
+      - Reference 1
+      - Reference 2
+    category: INJECTION  # info_gather, browser, server, misc, injection
```

Comment:
I think it would be more clear if the fields that can't be overridden were not present here. Was this to show that they are simply ignored? I think other test would be better if that's the case.

Reply:
Sorry, this was added because of copy / paste. Will remove. Nice catch.